include/drivers/gpio/gpio_sx1509b.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SX1509B_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SX1509B_H_
+
+int sx1509b_pwm_pin_configure(struct device *dev, gpio_pin_t pin);
```

Comment:
This needs documentation.  When it's configured to pwm mode, what state is it in?  How do you remove it from pwm mode back to normal GPIO?
 
Also the SX1509B provides generic LED functionality, not just PWM.  It's a shame to not provide access to all of the capability.

I don't see anything in the driver that prevents initiating standard gpio operations on a pin that's been configured for LED.

Reply:
Standard GPIO operations are prohibited when the LED driver is enabled.

I agree that all the SX1509B functionality should be exposed (both for the LED driver and keypad engine), but unfortunately I do not have time to implement this now.